analyses/ewings-cell-type/02-cellassign-tumor-cell-classification.Rmd

```diff
@@ -0,0 +1,641 @@
+We would expect to see expression of the marker genes for tumor cells to be higher in the cells that are identified to be tumor cells compared to other cell types. 
+
+```{r}
+grouped_celltype_df <- marker_gene_exp_df |> 
+  # group by barcode and cell type to get the sum 
+  dplyr::group_by(barcodes, cell_type) |> 
+  dplyr::mutate(mean_exp = mean(gene_expression))
+
+ggplot(grouped_celltype_df, aes(x = mean_exp, color = cell_type)) +
+  geom_density() +
+  facet_wrap(vars(visser_cellassign_annotations)) +
+  theme_bw() +
+  labs(
+    color = "Marker gene group"
+  )
+```
+
+It looks like the only real difference between groups is that the endothelial cells have higher expression of the tumor markers then other cells? 
+However the tumor cells do not show increased expression of the tumor marker genes. 
+This really is not what I would expect... maybe we need a larger reference... 
+
+
+### Run CellAssign with markers from PanglaoDB
+
+The last thing I'm trying here is to run CellAssign using some of the same reference that we used when running CellAssign as part of scpca-nf. 
+In particular, the majority of non-muscle cells (which we believe to be tumor cells) were identified as either Endothelial cells or Fibroblasts.
+So we will create a reference that takes the markers for Endothelial cells and Fibroblasts from PanglaoDB and combines with the tumor markers. 
+
+```{r}
+# get just the genes for endothelial and fibroblasts from panglao 
+panglao_markers_df <- readr::read_tsv(panglao_markers_file) |> 
+  dplyr::filter(`cell type` %in% c("Endothelial cells", "Fibroblasts")) |> 
+  dplyr::select(`official gene symbol`, `cell type`)
+
+# get the ensembl id from rowdata for each panglao marker 
+panglao_tumor_markers_df <- rowData(sce) |> 
+  as.data.frame() |> 
+  dplyr::filter(gene_symbol %in% panglao_markers_df$`official gene symbol`) |> 
+  dplyr::left_join(panglao_markers_df, by = c("gene_symbol" = "official gene symbol")) |> 
+  dplyr::select(gene_symbol, ensembl_gene_id = gene_ids, cell_type = `cell type`) |> 
+  # add in the tumor markers 
+  dplyr::bind_rows(tumor_markers_df)
+  
+```
+
+
+```{r}
+# build the reference matrix 
+panglao_tumor_ref_mtx <- build_binary_mtx(panglao_tumor_markers_df, 
+                                          gene_col = "ensembl_gene_id",
+                                          cell_type_col = "cell_type")
+
+# remove any genes that may be present in multiple cell types 
+rowsums <- panglao_tumor_ref_mtx |> 
+  dplyr::select(-ensembl_gene_id) |> 
+  rowSums()
+
+panglao_tumor_ref_mtx <- panglao_tumor_ref_mtx[-which(rowsums > 1),]
+
+# export matrix 
+readr::write_tsv(panglao_tumor_ref_mtx, panglao_markers_mtx_file)
+```
+
+
+```{r, eval=FALSE}
+cellassign_call <- glue::glue(
+  "python scripts/run-cellassign.py ",
+  "--anndata_file '{anndata_file}' ",
+  "--output_predictions '{panglao_predictions_file}' ",
+  "--reference '{panglao_markers_mtx_file}' "
```

Comment:
This one you had spaces at the end, so I would expect it to work, but I moved them to the beginning for consistency
```suggestion
  "python scripts/run-cellassign.py",
  " --anndata_file '{anndata_file}'",
  " --output_predictions '{panglao_predictions_file}'",
  " --reference '{panglao_markers_mtx_file}'"
```